corehq/tabs/uitab.py

```diff
@@ -127,7 +124,8 @@ def should_show(self):
         if not self.show_by_default and not self.is_active_tab:
             return False
 
-        if self.restrict_access_by_location and not self.filtered_dropdown_items:
```

Comment:
`self.restrict_access_by_location` was incorrectly evaluating to True because the users weren't yet members of the domain.  I've switched it to use `request.can_access_all_locations`, which handles that correctly.

That evaluating to True meant that `self.filtered_dropdown_items` to run.  Normally that runs in templates, which swallows the errors about `project == None`

Note that it doesn't error for logged-in users who are restricted.